src/runtime_src/core/tools/xbutil2/SubCmdReset.cpp

```diff
@@ -45,7 +46,39 @@ pretty_print_action_list(xrt_core::device_collection& deviceCollection, xrt_core
 static void
 reset_device(std::shared_ptr<xrt_core::device> dev, xrt_core::query::reset_type reset)
 {
-  dev->reset(reset);
+  if (xrt_core::device_query<xrt_core::query::rom_vbnv>(dev).find("_u30_") != std::string::npos) {
+    /*
+     * u30 reset relies on working SC and SN info. SN is read and saved
+     * when FPGA is ready. so even if there is firewall trip now, we expect
+     * to be able to get S/N again
+     * Having SN info available also implies there is a working SC
+     */
+    std::string sn;
+    try {
+      sn = xrt_core::device_query<xrt_core::query::xmc_serial_num>(dev);
+    } catch (std::exception& ex) {
+      std::cerr << "ERROR:" << ex.what() << std::endl;
+      return;
```

Comment:
Weird set of method exits.

Sometimes there is an exit early and other times not.  

Also with regards to the catch, is there a reason why it is being caught here as opposed to the calling method.

Reply:
we do throw an error in the calling method, but we should catch that error and cleanly print it out. If we don't catch the error, we get a "header" for the error as well. Unless it is an unexpected error, we don't need the error header.